version: '3'

services:
  db:
    image: docker.io/library/postgres
    volumes:
      - ./data/db:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB
      - POSTGRES_USER
      - POSTGRES_PASSWORD

  backend:
    build:
      context: ./Backend/backend
    command: gunicorn backend.wsgi:application --bind 0.0.0.0:8000 --workers 7 --worker-class gevent --timeout 180
    #command: python manage.py runserver 0.0.0.0:8000

    volumes:
      - .:/code
      - ./data/media:/media
    ports:
      - "8000:8000"
    environment:
      - POSTGRES_NAME=POSTGRES_DB
      - POSTGRES_USER
      - POSTGRES_PASSWORD
    depends_on:
      - db

  nginx:
    container_name: nginx
    image: nginx:latest
    restart: unless-stopped
    environment:
      - DOMAIN
    depends_on:
      - frontend
    ports:
      - 80:80
    volumes:
      - ./etc/nginx/templates:/etc/nginx/templates:ro

  certbot:
    container_name: certbot
    image: certbot/certbot:latest
    depends_on:
      - nginx
    command: >-
      certonly --reinstall --webroot --webroot-path=/var/www/certbot --email ${EMAIL} --agree-tos --no-eff-email -d ${DOMAIN}
    volumes:
      - ./etc/letsencrypt:/etc/letsencrypt
      - ./certbot/data:/var/www/certbot

  frontend:
    build:
      context: ./Frontend/frontend
    ports:
      - "80:80"
    environment:
      - VIRTUAL_HOST=na-mahlzeit.de,www.na-mahlzeit.de
      - LETSENCRYPT_HOST=na-mahlzeit.de,www.na-mahlzeit.de
      - LETSENCRYPT_EMAIL=fuerRo138@gmx.de
    depends_on:
      - backend

volumes:
  nginx_conf_ssl:
  letsencrypt_certs:
  certbot_acme_challenge:
  db-volume:
